cognite/client/utils/_pyodide_helpers.py

```diff
@@ -0,0 +1,218 @@
+
+    def submit(self, fn: Callable[..., T_Result], *args: Any, **kwargs: Any) -> DummyFuture:
+        if "priority" in inspect.signature(fn).parameters:
+            raise TypeError(f"Given function {fn} cannot accept reserved parameter name `priority`")
+        kwargs.pop("priority", None)
+        return DummyFuture(fn, *args, **kwargs)
+
+    def shutdown(self, wait: bool = False) -> None:
+        return None
+
+    def __enter__(self) -> SerialPriorityThreadPoolExecutor:
+        return self
+
+    def __exit__(
+        self,
+        type: Optional[type[BaseException]],
+        value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        self.shutdown()
+
+
+class EnvVarToken(CredentialProvider):
+    """Credential provider that always reads token from an environment variable just-in-time,
+    allowing refreshing the value by another entity.
+    Args:
+        key (str | None): The name of the env.var. to read from. Default: 'COGNITE_TOKEN'
+    Raises:
+        KeyError: If the env.var. is not set.
+    """
+
+    def __init__(self, key: str = None) -> None:
+        self.key = "COGNITE_TOKEN" if key is None else key
+
+    def __token_factory(self) -> str:
+        return os.environ[self.key]
+
+    def authorization_header(self) -> Tuple[str, str]:
+        return "Authorization", f"Bearer {self.__token_factory()}"
+
+
+class OAuthInteractiveFusionNotebook(OAuthInteractive):
+    # TODO: PublicClientApplication is broken in pyodide because of requests?
+    def __init__(self, redirect_port: int = 53000) -> None:
+        token, base_url = os.environ["COGNITE_TOKEN"], os.environ["COGNITE_BASE_URL"]
+        payload = self._load_payload_from_jwt(token)
+        client_id = payload["appid"]
+        authority_url = f"https://login.microsoftonline.com/{payload['tid']}"
+
+        # Note: Death to name mangling:
+        self._OAuthCredentialProviderWithTokenRefresh__access_token = token
+        self._OAuthCredentialProviderWithTokenRefresh__token_refresh_lock = contextlib.nullcontext()
+        self._OAuthCredentialProviderWithTokenRefresh__access_token_expires_at = payload["exp"]
+
+        self._OAuthInteractive__authority_url = authority_url
+        self._OAuthInteractive__client_id = client_id
+        self._OAuthInteractive__scopes = [f"{base_url}/.default"]
+        self._OAuthInteractive__redirect_port = redirect_port
+        self._OAuthInteractive__app = PublicClientApplication(client_id=client_id, authority=authority_url)
+
+    @staticmethod
+    def _load_payload_from_jwt(token: str) -> Dict[str, Any]:
+        """A JWT has the structure: `<header>.<payload>.<signature>`. This method returns payload as a dict"""
+        _, payload, _ = token.split(".")
+        if remainder := len(payload_bytes := payload.encode()) % 4:
+            payload_bytes += b"=" * (4 - remainder)
+        return json.loads(urlsafe_b64decode(payload_bytes))
```

Comment:
IMO if you're using the variable outside the scope of the if-statement, it's easier to follow if you define it upfront.
```suggestion
        payload_bytes = payload.encode()
        if remainder := len(payload_bytes) % 4:
            payload_bytes += b"=" * (4 - remainder)
        return json.loads(urlsafe_b64decode(payload_bytes))
```

Reply:
Entire class removed, see other comment